QPong-Unity-Version/Assets/Scripts/CircuitGridClient.cs

```diff
@@ -27,70 +31,97 @@ void Start()
-            stateProbability[i] = Complex.Pow(stateVector[i], 2).Magnitude;
-            paddleArray[i].GetComponent<SpriteRenderer> ().color = new Color (1,1,1,(float)stateProbability[i]);
-        }
-        Debug.Log("State Probability: ["+string.Join(", ", stateProbability)+"]");
+        Debug.Log("Send Gate Array: "+ gateString);
+        string urlString = API_URL + API_VERSION + "get_statevector";
+        StartCoroutine(PostRequest(urlString, gateString, (results) => {
+
+            // Deserialize stateVector from JSON
+            //TODO: come up with a better way to abstract this out
+            var obj = JsonConvert.DeserializeObject<RootObject>(results);
+            Complex[] stateVector = new Complex[numberOfStates];
+            double[] stateProbability = new double[numberOfStates];
+            paddleArray = GameObject.Find("CircuitGrid").GetComponent<CircuitGridControl>().paddleArray;
+            for (int i = 0; i < numberOfStates; i++)
+            {
+                stateVector[i] = new Complex(obj.__ndarray__[i].__complex__[0], obj.__ndarray__[i].__complex__[1]);
+                stateProbability[i] = Complex.Pow(stateVector[i], 2).Magnitude;
+                paddleArray[i].GetComponent<SpriteRenderer>().color = new Color(1, 1, 1, (float)stateProbability[i]);
+            }
+           // Debug.Log("State Probability: [" + string.Join(", ", stateProbability) + "]");
+        }));
+    }
+
+    IEnumerator DelayStateVector(string gatesString, float delay)
+    {
+        yield return new WaitForSeconds(delay);
+
+        GetStateVector(gatesString);
+    }
+
+    private void DoMeasurement(string gatesString)
+    {
+        Debug.Log("Send Gate Array: "+ gatesString);
+        string urlString = API_URL + API_VERSION + "do_measurement";
+        StartCoroutine(PostRequest(urlString, gatesString, (results) =>
+        {
+            paddleArray = GameObject.Find("CircuitGrid").GetComponent<CircuitGridControl>().paddleArray;
+            for (int i = 0; i < 8; i++)
+            {
+                // make all states invisible and disable colliders
+                paddleArray[i].GetComponent<SpriteRenderer>().color = new Color(1, 1, 1, 0);
+                paddleArray[i].GetComponent<BoxCollider2D>().enabled = false;
+            }
+            int stateInDecimal = Int32.Parse(results);
+            // make the measured state visible and enable collider
+            paddleArray[stateInDecimal].GetComponent<SpriteRenderer>().color = new Color(1, 1, 1, 1);
+            paddleArray[stateInDecimal].GetComponent<BoxCollider2D>().enabled = true;
+
+            //// Show statevector representation again sometime after measurement
+            StartCoroutine(DelayStateVector(gatesString, 0.3f));
+        }));
+
     }
 
-    IEnumerator doMeasurement(string gateArrayString)
+    public IEnumerator PostRequest(string uri, string gatesString, Action<string> completionHandler)
     {
```

Comment:
Is there a typo in "uri"? Should it be "url"?

Reply:
uri wasn't a typo, it seems to be how Unity calls it, as the PostRequest method is modeled after their documentation.  Uri stands for Universal Resource Identifier. and the L in URL is locator.  I had to look that up!  I don't really care what we call it so if you want url, you got it, will change